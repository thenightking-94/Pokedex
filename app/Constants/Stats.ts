export const stats = ['HP', 'Attack', 'Defense', 'Sp. Atk', 'Sp. Def', 'Speed']