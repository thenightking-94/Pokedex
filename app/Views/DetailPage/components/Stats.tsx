import { Grid } from '@mui/material';
import React, { ReactElement } from 'react';
import { stats } from '../../../Constants';

export default function Stats ({detailData}): ReactElement{
    return(
        <div className='stats'>
            <div className='title'>
                {'Stats'}
            </div>
            <Grid className='stats-grid' container spacing={2}>
                {detailData?.stats?.map((item, index) => {
                    return (
                        <Grid container key={index} xs={12} item sm={6}>
                            <Grid item xs={3} sm={3} className='label'>{stats[index]}</Grid>
                            <Grid item xs={9} sm={9} className='label-seek'>
                                <div style={{width:`${item.base_stat > 100? 100 : item.base_stat}%`}} className='filler'>{item.base_stat}</div>
                            </Grid>
                        </Grid>
                    )
                })}
            </Grid>
     </div>

    )

}