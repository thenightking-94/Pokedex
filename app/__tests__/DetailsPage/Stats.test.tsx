import { render } from '@testing-library/react'
import { QueryClient, QueryClientProvider } from 'react-query'
import renderer from "react-test-renderer";
import Stats from '../../Views/DetailPage/components/Stats'

const queryClient = new QueryClient()

const props = { detailData: {} }

describe('Stats should render', () => {
  it('renders correctly', () => {
    render(
    <QueryClientProvider client={queryClient}>
         <Stats {...props} />
    </QueryClientProvider>
   )
  })
})

describe("Jest Snapshot testing suite", () => {
    it("Matches DOM Snapshot", () => {
      const domTree = renderer.create( 
      <QueryClientProvider client={queryClient}>
        <Stats {...props}/>
     </QueryClientProvider>).toJSON();
      expect(domTree).toMatchSnapshot();
    });
  });