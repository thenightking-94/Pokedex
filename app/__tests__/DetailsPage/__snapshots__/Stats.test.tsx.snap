// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Jest Snapshot testing suite Matches DOM Snapshot 1`] = `
<div
  className="stats"
>
  <div
    className="title"
  >
    Stats
  </div>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 stats-grid css-mhc70k-MuiGrid-root"
  />
</div>
`;
